pkg/config/zonepb/zone.go

```diff
@@ -1094,6 +1094,75 @@ func (z *ZoneConfig) LeasePreference(i int) cat.ConstraintSet {
 	return &z.LeasePreferences[i]
 }
 
+func (z *ZoneConfig) ToSpanConfig() (roachpb.SpanConfig, error) {
+	var sc roachpb.SpanConfig
+	// Copy over the values.
+	sc.RangeMinBytes = *z.RangeMinBytes
+	sc.RangeMaxBytes = *z.RangeMaxBytes
+	sc.GCTTL = z.GC.TTLSeconds
+
+	// GlobalReads is false by default.
+	sc.GlobalReads = false
+	if z.GlobalReads != nil {
+		sc.GlobalReads = *z.GlobalReads
+	}
+	sc.NumReplicas = *z.NumReplicas
+
+	// NumVoters = NumReplicas if NumVoters is unset.
+	sc.NumVoters = *z.NumReplicas
```

Comment:
This is incorrect. SpanConfig.NumVoters has the same semantics as zonepb.ZoneConfig.NumVoters, i.e. if it contains the zero-value, look at NumReplicas. We shouldn't set it to NumReplicas explicitly ourselves.